### PR TITLE
Add lifetimes recipe

### PR DESCRIPTION
**Pull request by Arnoldosmium**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/lifetimes/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+requirements:
+  host:
+    - autograd >=1.2.0
+    - dill >=0.2.6
+    - numpy >=1.10.0
+    - pandas >=0.24.0
+    - pip
+    - python
+    - scipy >=1.0.0
+    - setuptools
```
Comment (scopatz) [inline recipes/lifetimes/meta.yaml]: Do you really need all of these just to install?

File: recipes/lifetimes/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  run:
+    - autograd >=1.2.0
+    - dill >=0.2.6
+    - numpy >=1.10.0
+    - pandas >=0.24.0
+    - python
+    - scipy >=1.0.0
+    - setuptools
```
Comment (scopatz) [inline recipes/lifetimes/meta.yaml]: Do you need setuptools at runtime?

File: recipes/lifetimes/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    - lifetimes.datasets
+
+about:
+  home: https://github.com/CamDavidsonPilon/lifetimes
+  license: MIT
+  license_family: MIT
+  license_file: README.md
+  summary: Measure customer lifetime value in Python
+  doc_url: https://lifetimes.readthedocs.io/en/latest/?badge=latest
+  dev_url: ''
```
Comment (scopatz) [inline recipes/lifetimes/meta.yaml]: Please remove empty entries

File: recipes/lifetimes/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+extra:
+  recipe-maintainers:
+    - CamDavidsonPilon
```
Comment (scopatz) [inline recipes/lifetimes/meta.yaml]: @CamDavidsonPilon are you OK being a maintainer?

File: recipes/lifetimes/meta.yaml (hunk 2 above)
Comment (Arnoldosmium) [reply recipes/lifetimes/meta.yaml]: Yes. A sub-package initialization depends on a function in `setuptools`. Removing this fails the test.

Comment (scopatz) [reply recipes/lifetimes/meta.yaml]: Ok!

File: recipes/lifetimes/meta.yaml (hunk 4 above)
Comment (CamDavidsonPilon) [reply recipes/lifetimes/meta.yaml]: Yup, 👌with me

**Comment by scopatz** [comment]
Let us know when this is ready to go

**Comment by Arnoldosmium** [comment]
@scopatz Love to get this merged if the appveyor failed check is a global issue atm.

**Pull request merged**